File: Data Structures & Algorithms/jump-game/submission-5.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        if (nums.size() <= 1) return true;
        
        int val = nums.size() - 1;
        
        for (int i = nums.size() - 2; i >= 0; i--) {
            if (i + nums[i] >= val) {
                val = i;
            }
        }
        return val == 0;
    }
};

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-5.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        memo = [1] * len(nums)
        for i in range(1, len(nums)):
            for j in range(i - 1, -1, -1):
                if nums[i] > nums[j]:
                    memo[i] = max(memo[i], memo[j] + 1)
        return max(memo)
